### DISCUSSION TITLE
Support for sorting by disc and track number tags

### DISCUSSION THREAD

**Discussion by durin42** [music-assistant/support#3164] [Q&A] [answered]
I have a number of albums which are multi-disc sets, and right now Music Assistant only sorts by track number, so I end up with this type of play order:
```
Disc 2 Track 1
Disc 1 Track 1
Disc 1 Track 2
Disc 2 Track 2
...
```

Where the relative order of discs 1 and 2 is typically random for each track. It'd be awesome to have support for the disc number field as well as the track number.

**Comment by marcelveldt** [comment]
We sort by disc and track number so if that isnt working that is a bug

**Reply by durin42** [reply]
I know the metadata exists because I can see it in navidrome as being multi-disc. Various subsonic players on my phone also sort with the disc number, so navidrome seems to be exporting it via the API - could it be a bug in the subsonic component in Music Assistant? How would I debug that? (I don't see track numbers _or_ disc numbers in the Music Assistant UI, though I could be missing something obvious.)

**Reply by OzGav** [reply]
@khers

**Reply by khers** [reply]
What version of Music Assistant are you using? Release 2.3.3 had a fix for disc numbering but it could be that Navidrome is doing it differently. Are you able to use https://github.com/khers/py-opensonic to interact with your Navidrome server directly?

**Reply by durin42** [reply, marked as answer]
Upgrading to 2.3.4 (I was on 2.3.0) didn't fix it - as an experiment I disabled the Subsonic source and re-enabled it but that didn't help. Then as a final guess before diving in with a jupyter notebook and py-opensonic I:
 * disabled the subsonic source
 * stopped music-assistant
 * deleted the `library.db` (after backing up)
 * started music-assistant
 * enabled the subsonic source

Now I see the disc-number fields, so it looks like something was borked in the database. It's fixed! Sorry for the extra work on your end, if any.

Is there some more graceful thing I could have done to force an update of the library entries? Nuking the sqlite db seems like a large hammer.

Thanks!

**Reply by OzGav** [reply]
Yes. MA SETTINGS >> CORE >> MUSIC CONTROLLER >> ADVANCED >> RESET DATABASE

**Reply by durin42** [reply]
Awesome, thanks!
